subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheLifecyclePluginIntegrationTest.groovy

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache
+
+import groovy.transform.CompileStatic
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters
+
+class ConfigurationCacheLifecyclePluginIntegrationTest  extends AbstractConfigurationCacheIntegrationTest {
+
+    def 'buildDirectory is finalized when writing to the cache'() {
+        given:
+        def configurationCache = newConfigurationCacheFixture()
+        def buildDirName = 'my-build-dir'
+        def buildDir = file(buildDirName)
+        buildFile """
+
+            plugins { id 'lifecycle-base' }
+
+            @$CompileStatic.name
```

Comment:
Is this to avoid an import or fully qualifying, or something else? Either would read better IMO.

Reply:
This is to import it in the file and then reuse the imported reference in the test script.

Reply:
I don't get the benefit though. Also, not clear why we are using the annotation at all - the use of interpolation makes it just more distracting.

Reply:
The benefit is code navigation, find references, quick documentation, etc, will recognize that reference. I tend to favor type based navigation over finding strings and keeping those references usually help me in navigating the code.

Reply:
So the above is the rationale for using type interpolation. It doesn't explain why the annotation is there in the 1st place, the reason for that is that I was using the reproducer as provided, I'll simplify it.

Reply:
I've simplified the test.